contracts/utils/DAOTracker.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.5.4;
+
+import "@daostack/infra/contracts/Reputation.sol";
+import "../controller/DAOToken.sol";
+import "../controller/Avatar.sol";
+import "../controller/ControllerInterface.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+/**
+ * @title An on-chain "source of truth" for what DAOs
+ *        should be index into DAOstack's subgraph.
+ */
+contract DAOTracker is Ownable {
+
+    struct TrackingInfo {
+        DAOToken            nativeToken;
+        Reputation          nativeReputation;
+        ControllerInterface controller;
+
+        // `blacklist` this from the subgraph's cache
+        bool blacklist;
+    }
+
+    // Mapping from Avatar address to a flag letting the subgraph
+    // know if it should track this DAO or not.
+    mapping(address=>TrackingInfo) public tracking;
+
+    event TrackDAO(address indexed _avatar, address indexed _controller);
+
+    modifier onlyAvatarOwner(Avatar avatar) {
+        require(avatar.owner() == msg.sender,
+                "The caller must be the owner of the Avatar.");
+        _;
+    }
+
+    /**
+    * @dev track a new organization. This function will tell the subgraph
+    *      to start ingesting events from the DAO's contracts.
+    *      NOTE: This function should be called as early as possible in the DAO deployment
+    *      process. **Smart Contract Events that are emitted from blocks prior to this function's
+    *      event being emitted WILL NOT be ingested into the subgraph**, leading to an incorrect
+    *      cache. If this happens to you, please contact the subgraph maintainer. Your DAO will
+    *      need to be added to the subgraph's startup config, and the cache will need to be rebuilt.
+    * @param _avatar the organization avatar
+    * @param _controller the organization controller
+    */
+    function track(Avatar _avatar, ControllerInterface _controller)
+    public
+    onlyAvatarOwner(_avatar) {
+        // Only allow the information to be set once. In the case of a controller upgrades,
+        // the subgraph will be updated via the UpgradeController event.
```

Comment:
should we add a daoTrack event upon UpgradeController event ?

Reply:
Not sure what you mean by this?

Reply:
it will be easier for the subgraph to handle one event (TraclDAP) .so by emitting TrackDAO event from the UpgradeController function the subgraph will be happier.

Reply:
I view these as two completely separate cases requiring two sets of independent logic. I think it's valuable to keep them separate so the subgraph mappings can choose what it wants to do.

Reply:
Please let me know if this isn't the case ^^^